Core/clim-core/commands/tables.lisp

```diff
@@ -407,77 +393,76 @@ examine the type of the command menu item to see if it is
-    (with-slots (menu) table
-      (if (and errorp (not item))
-          (error 'command-not-present :command-table-name
-                 (command-table-designator-as-name table))
+  (setf table (find-command-table table))
+  (with-slots (menu) table
+    (if-let ((item (find-menu-item string table :errorp nil)))
+      ;; When a keystroke is still present leave the item be.
+      (if (null (command-menu-item-keystroke item))
           (setf menu (delete string menu
                              :key #'command-menu-item-name
-                             :test #'string-equal))))))
+                             :test #'string-equal))
+          (setf (command-menu-item-name item) nil))
+      (when errorp
+        (error 'command-not-present :command-table-name
+               (command-table-designator-as-name table))))))
 
 (defun map-over-command-table-keystrokes
     (function command-table &key (inherited t))
-  (declare (ignore inherited))
-  (let ((command-table (find-command-table command-table)))
-    (with-slots (keystroke-accelerators keystroke-items)
-        command-table
-      (loop for gesture in keystroke-accelerators
-            for item in keystroke-items
-            do (funcall function
-                        (command-menu-item-name item)
-                        gesture
-                        item)))))
-
-(defun find-keystroke-item (gesture command-table
+  (let ((table-object (find-command-table command-table)))
+    (flet ((map-table-entries (table)
+             (mapc #'(lambda (item)
+                       (with-slots (menu-name keystroke) item
+                         (when keystroke
+                           (funcall function menu-name keystroke item))))
+                   (slot-value table 'menu))))
+      (map-table-entries table-object)
+      (when (and inherited (inherit-keystrokes table-object))
+        (dolist (table (command-table-inherit-from table-object))
+          (map-over-command-table-keystrokes function table))))
+    (values)))
+
+(defun find-keystroke-item (gesture table
                             &key (test #'event-matches-gesture-name-p)
-                            (errorp t))
-  (let ((command-table (find-command-table command-table)))
-    (loop for keystroke in (slot-value command-table 'keystroke-accelerators)
-          for item in (slot-value command-table 'keystroke-items)
-          if (funcall test gesture keystroke)
-          do (return-from find-keystroke-item (values item command-table)))
-    (if errorp
-        (error 'command-not-present :command-table-name
-               (command-table-designator-as-name command-table))
-        nil)))
+                              (errorp t))
+  (setf table (find-command-table table))
+  (loop for item in (slot-value table 'menu)
+        when (funcall test gesture (command-menu-item-keystroke item))
+          do (return-from find-keystroke-item
+               (values item table)))
```

Comment:
Why `loop` instead of `find`?

Reply:
I don't have anything against find, however

```
(when-let ((item (find gesture (slot-value table 'menu)
                       :test test
                       :key #'command-menu-item-keystroke)))
  (return-from find-keystroke-item
    (values item table)))
```

does not look /much/ better. I'd probably write it with find now, but this code is removed one commit later and is not incorrect, so I'd rather leave it be.

Reply:
Right, it's mood since replaced in subsequent commit.